v2/options.go

```diff
@@ -0,0 +1,48 @@
+package car
+
+type ReadOptions struct {
+	V1NullPadding bool
+
+	BlockstoreUseWholeCIDs bool
+}
+
+type ReadOption func(*ReadOptions)
+
+func (ReadOption) readWriteOption() {}
+
+var _ ReadWriteOption = ReadOption(nil)
+
+type WriteOptions struct {
+	CarV1Padding uint64
+	IndexPadding uint64
+
+	BlockstoreAllowDuplicatePuts bool
+}
+
+type WriteOption func(*WriteOptions)
+
+func (WriteOption) readWriteOption() {}
+
+var _ ReadWriteOption = WriteOption(nil)
+
+type ReadWriteOption interface {
+	readWriteOption()
+}
+
+func AllowV1NullPadding(o *ReadOptions) {
+	o.V1NullPadding = true
+}
+
+// UseCarV1Padding sets the padding to be added between CAR v2 header and its data payload on Finalize.
+func UseCarV1Padding(p uint64) WriteOption {
+	return func(o *WriteOptions) {
+		o.CarV1Padding = p
+	}
+}
+
+// UseIndexPadding sets the padding between data payload and its index on Finalize.
+func UseIndexPadding(p uint64) WriteOption {
```

Comment:
I put the padding opts at the root, because they're not specific to the blockstore, and because we might want to use them later on for the Writer API. But I'm not 100% sure on this.